common/models/templates/src/trie-model.ts

```diff
@@ -161,6 +161,116 @@
     public wordbreak(context: Context): USVString {
       return this.getLastWord(context.left);
     }
+
+    public getRootTraversal(): LexiconTraversal {
+      return new TrieModel.Traversal(this._trie['root'], '');
+    }
+
+    private static Traversal = class implements LexiconTraversal {
+      prefix: String;
+      root: Node;
+
+      constructor(root: Node, prefix: string) {
+        this.root = root;
+        this.prefix = prefix;
+      }
+
+      *children(): Generator<{char: string, traversal: () => LexiconTraversal}> {
+        let root = this.root;
+
+        if(root.type == 'internal') {
+          for(let i = 0; i < root.values.length; i++) {
+            let entry = root.values[i];
+            let entryNode = root.children[entry];
+
+            // SMP check
```

Comment:
Pedantic note: "SMP" standard for the [Supplementary Multilingual Plane](https://en.wikipedia.org/wiki/Plane_(Unicode)#Supplementary_Multilingual_Plane), which is U+010000–U+01FFFF. This check is for finding a surrogate pair, which may encode any code point outside of the Basic Multilingual Plane (U+0000–U+FFFF).

All code points U+010000 and above don't officially have a better name than "non-BMP", but I think they're colloquially called the "[astral planes](http://www.opoudjis.net/unicode/unicode_astral.html)", as UTF-16 must astral project to encode code points in the range 😂

Reply:
So the check is (mostly) right, but it's the nomenclature that's wrong?

Reply:
Correct! Just use "non-BMP" or "astral code points" and it's all good :)